### PR TITLE
Update interceptors.js

### PR DESCRIPTION
**Pull request by marknokes**
add error handling to fullResponse for simplified troubleshooting

### CODE REVIEW

**Comment by knicola** [comment]
Thank you for taking time to contribute and I apologize for the delay! I'll be reviewing this in a bit.

**Comment by marknokes** [comment]
My pleasure! Thank you for your great work on this. I've been building an automated strategy using your library and it's been really fun!

**Review by knicola — CHANGES_REQUESTED** [review]
Besides the change requested in the comment above, it is always nice to have a test accompanying any proposed changes to the code for coverage. You can look at this test as an example to get you started: https://github.com/knicola/tdameritradejs/blob/097f5dab7cf72cacfe3a81ebad8e3a377a1586f3/tests/client.spec.js#L903
However, if you are not sure about the test let me know and I can take care of it. Thanks!

**Comment by knicola** [comment]
> My pleasure! Thank you for your great work on this. I've been building an automated strategy using your library and it's been really fun!

I'm really glad to hear! It's always nice to get positive feedback. All the best with your strategy!

**Comment by marknokes** [comment]
Yes, of course! I'm sorry for the hasty pull request. I have updated for consistency. I wasn't sure about how to perform the test. Please let me know if my latest changes look okay. For the error response when `client.config.returnFullResponse` is set to false, I was attempting to only return what I considered the most significant information (the status and message). I have added the lodash get method to retrieve the values from the error response object.

File: src/client/interceptors.js [hunk 1]
```diff
@@ -81,6 +81,11 @@ function fullResponse(client) {
         return client.config.returnFullResponse
             ? response
             : get(response, 'data')
+    }, error => {
+        return {
```
[the remark this answers is not in the document]
Comment (marknokes) [reply src/client/interceptors.js]: Yes, of course! I'm sorry for the hasty pull request. I have updated for consistency. I wasn't sure about how to perform the test. Please let me know if my latest changes look okay. For the error response when client.config.returnFullResponse is set to false, I was attempting to only return what I considered the most significant information (the status and message). I have added the lodash get method to retrieve the values from the error response object.

[the remark this answers is not in the document]
Comment (knicola) [reply src/client/interceptors.js]: No worries! I'll write the test over the weekend before merging but for now I believe we're good. Much appreciated!

**Comment by knicola** [comment]
Hello, apologies for the long delay. Life got a bit busy.

While writing the tests, I realized that the "error" needs to be thrown rather than returned otherwise it'll be considered as handled and will never trigger `.catch()`. Also, we need to take into account any errors that do not contain a server response (ie. a network error).

I will not be using this PR, as I see you've pushed more commits that are not related to the original issue, but I do have a fix on the way and you will be added as a co-author.

PS. I'll be happy to discuss the additional commits, if that is something you are interested in. Feel free to open a new issue or even a [gh discussion](https://github.com/knicola/tdameritradejs/discussions/categories/ideas).

**Pull request closed without merging**